initfiles/sbin/complete-uninstall.sh.in

```diff
@@ -88,24 +135,30 @@ elif [ -e /etc/redhat-release -o -e /etc/SuSE-release ]; then
 fi
 
 echo "Removing Directory - ${path}"
-rm -rf ${path}
+removedir ${path}
+
+if [ -n "${mklink}" -a -n "${lpath}" ] ; then
```

Comment:
A comment on why this is done this way would be useful.

Is recreating the directory/softlink good enough? Do you need to worry about preserving access rights/permissions/ownership?